models/classes/runner/time/TimerAdjustmentService.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA ;
+ */
+
+namespace oat\taoQtiTest\models\runner\time;
+
+use oat\oatbox\service\ConfigurableService;
+use oat\taoDelivery\model\execution\DeliveryExecutionInterface;
+use oat\taoQtiTest\models\runner\StorageManager;
+use oat\taoQtiTest\models\TestSessionService;
+use oat\taoTests\models\runner\time\TimerAdjustmentServiceInterface;
+use qtism\data\QtiIdentifiable;
+
+class TimerAdjustmentService extends ConfigurableService implements TimerAdjustmentServiceInterface
+{
+    /** @var QtiTimer $timer */
+    private $timer;
+
+    /**
+     * @inheritDoc
+     */
+    public function increase($deliveryExecution, $seconds, $source = null)
+    {
+        $this->timer = $this->getTestSessionService()->getTestSession($deliveryExecution)->getTimer();
```

Comment:
Probably it would be better to get timer inside of `register` method where it's used and pass it as a parameter to `findMaximumPossibleDecrease()` and `putAdjustmentToTheMap()` methods to make code a bit more clear.

Reply:
I agree it wasn't very clear, I've updated the implementation. Yet I still use a class variable for `deliveryExecution` in order to avoid passing it around.